fair_mango/metrics/metrics.py

```diff
@@ -697,3 +700,280 @@ def is_biased(self, threshold: float = 0.1) -> dict[str, bool]:
                 bias[target] = False
 
         return bias
+
+
+def super_set_fairness_metrics(
+    metric: (
+        type[DemographicParityDifference]
+        | type[DemographicParityRatio]
+        | type[DisparateImpactDifference]
+        | type[DisparateImpactRatio]
+        | type[EqualOpportunityDifference]
+        | type[EqualOpportunityRatio]
+        | type[EqualisedOddsDifference]
+        | type[EqualisedOddsRatio]
+        | type[FalsePositiveRateDifference]
+        | type[FalsePositiveRateRatio]
+    ),
+    data: Dataset | pd.DataFrame,
+    sensitive: Sequence[str] = [],
```

Comment:
nope nope nope, we talked about this

Reply:
I initialised the value to `None`.

Reply:
It's not the first time that you made the mistake, so here is a little exercise :

without using chatGPT, could you give me examples why setting a list as a default value is probably not what we want ? Make a small paragraph explaining how lists are handled behind the scenes etc.

Reply:
Setting a mutable as the default value like setting a parameter to a list will not ensure using the initial value of the list every time the function is called. Instead the updated version of the list is used (if anything is appended or removed, it will be used with the changes). 
Example:
```python
def append_to_list(l = []):
    l.append(1)
    print(l)

append_to_list()
append_to_list()
append_to_list()
```
Output:
```
[1]
[1, 1]
[1, 1, 1]
```

Reply:
why this happens ? What else behaves like that ?

Reply:
Because in Python, the default values are evaluated once and not every time the function is called. And this happens for every mutable object (lists, sets, dict..). I believe this is because the variable holds a reference to their memory location and not their actual value.